packages/design-system/src/components/SimpleMenu/Menu.tsx

```diff
@@ -24,23 +25,50 @@ const MenuRoot = DropdownMenu.Root;
  * MenuTrigger
  * -----------------------------------------------------------------------------------------------*/
 
-interface TriggerProps extends ButtonProps {}
+type TriggerPropsBase = Omit<ButtonProps, 'tag' | 'children'> & {
+  endIcon?: React.ReactNode;
+  label?: React.ReactNode | string;
+};
+
+type TriggerPropsWithButton = TriggerPropsBase & {
+  tag?: typeof Button;
+  icon?: React.ReactNode;
+};
+
+type TriggerPropsWithIconButton = TriggerPropsBase & {
+  tag: typeof IconButton;
+  icon: React.ReactNode;
+};
+
+type TriggerProps = TriggerPropsWithButton | TriggerPropsWithIconButton;
 
 const MenuTrigger = React.forwardRef<HTMLButtonElement, TriggerProps>(
-  ({ size, endIcon = <CaretDown width="0.6rem" height="0.4rem" aria-hidden />, ...props }, ref) => {
+  (
+    { label, size, endIcon = <CaretDown width="0.6rem" height="0.4rem" aria-hidden />, tag = Button, icon, ...rest },
```

Comment:
Could you double check that the caret icon is the right size? It looks really tiny in the storybook example

Reply:
yeah..updated!